direct-auth-samples/src/test/java/info/seleniumcucumber/userStepDefinitions/Login.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2021-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package info.seleniumcucumber.userStepDefinitions;
+
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import env.CucumberRoot;
+import env.DriverUtil;
+
+import org.junit.Assert;
+import org.openqa.selenium.WebDriver;
+import pages.ForgotPasswordPage;
+import pages.LoginPage;
+import pages.RootPage;
+
+public class Login extends CucumberRoot {
+
+	protected WebDriver driver = DriverUtil.getDefaultDriver();
+	private RootPage rootPage = new RootPage(driver);
+	private LoginPage loginPage = new LoginPage(driver);
+	private ForgotPasswordPage forgotPasswordPage = new ForgotPasswordPage(driver);
+
+	@When("^she fills in her correct username$")
+	public void enter_correct_username() {
+		loginPage.sleep();
+		loginPage.usernameInput.click();
+		loginPage.usernameInput.sendKeys(USERNAME);
+	}
+
+	@And("^she fills in her correct password$")
+	public void enter_correct_password() {
+		loginPage.passwordInput.click();
+		loginPage.passwordInput.sendKeys(PASSWORD);
+	}
+
+	@And("^she submits the Login form$")
+	public void clicks_login_button() {
+		loginPage.signInButton.click();
+	}
+
+	@Then("^she is redirected to the Root View$")
+	public void redirected_to_root_view() {
+		rootPage.sleep();
+		Assert.assertTrue(rootPage.email.isDisplayed());
+		String email = rootPage.email.getText();
+		Assert.assertFalse("Can't access profile information", email.isEmpty());
+	}
+
+	@And("^the access_token is stored in session$")
+	public void access_token_stored() {
+		rootPage.sleep();
+		Assert.assertTrue(rootPage.accessToken.isDisplayed());
+		String accessToken = rootPage.accessToken.getText();
+		Assert.assertFalse("Can't access access_token", accessToken.isEmpty());
+	}
+
+	@And("^the id_token is stored in session$")
+	public void id_token_stored() {
+		rootPage.sleep();
+		Assert.assertTrue(rootPage.idToken.isDisplayed());
+		String idToken = rootPage.idToken.getText();
+		Assert.assertFalse("Can't access id_token", idToken.isEmpty());
+	}
+
+	@And("^the refresh_token is stored in session$")
```

Comment:
refresh_token is not stored if `offline_access` is not added in the `OKTA_IDX_SCOPES` variable.
It should pass on CI. Please test locally if possible by adding `offline_access` to the scopes.